frontend/src/component/common/FilterDateItem/FilterDateItem.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { screen } from '@testing-library/react';
+import { render } from 'utils/testRenderer';
+import { vi } from 'vitest';
+import { FilterItemParams } from '../FilterItem/FilterItem';
+import { FilterDateItem, IFilterDateItemProps } from './FilterDateItem';
+
+const getDate = (option: string) => screen.getByText(option);
+
+const setup = (initialState: FilterItemParams) => {
+    const recordedChanges: FilterItemParams[] = [];
+    const mockProps: IFilterDateItemProps = {
+        label: 'Test Label',
+        onChange: (value: FilterItemParams) => {
+            recordedChanges.push(value);
+        },
+        onChipClose: () => {},
+        operators: ['IS_ON_OR_AFTER', 'IS_BEFORE'],
+        state: initialState,
+    };
+
+    render(<FilterDateItem {...mockProps} />);
+
+    return recordedChanges;
+};
+
+afterEach(() => {
+    vi.restoreAllMocks();
+});
+
+describe('FilterDateItem Component', () => {
+    it('renders initial state correctly', async () => {
+        vi.mock('hooks/useLocationSettings', () => ({
```

Comment:
import overrides are very hacky and magical as they overwrite how native imports work. Can we handle it similar to how we do local storage?

Reply:
Actually this was a idea I was working on and left in. Removed.

Reply:
overall I'd recommend not using jest.someMagic or vi.somMagic. It usually means the code is not testable and requires some testing framework magic.